android/src/main/java/com/reactnativecommunity/imageeditor/ImageEditorModuleImpl.kt

```diff
@@ -318,14 +326,28 @@ class ImageEditorModuleImpl(private val reactContext: ReactApplicationContext) {
         return Bitmap.createBitmap(bitmap, cropX, cropY, cropWidth, cropHeight, scaleMatrix, filter)
     }
 
-    private fun openBitmapInputStream(uri: String): InputStream? {
+    private fun openBitmapInputStream(uri: String, headers: ReadableMap?): InputStream? {
         return if (uri.startsWith("data:")) {
             val src = uri.substring(uri.indexOf(",") + 1)
             ByteArrayInputStream(Base64.decode(src, Base64.DEFAULT))
         } else if (isLocalUri(uri)) {
             reactContext.contentResolver.openInputStream(Uri.parse(uri))
         } else {
             val connection = URL(uri).openConnection()
+            if (headers != null) {
+                val iterator = headers.keySetIterator()
+                while (iterator.hasNextKey()) {
+                    val key = iterator.nextKey()
+                    val value: String? =
+                        when (headers.getType(key)) {
+                            ReadableType.String -> headers.getString(key)
+                            else -> null
+                        }
+                    if (value != null) {
+                        connection.setRequestProperty(key, value)
+                    }
+                }
```

Comment:
I'd also suggest to do conversion to `HashMap` already in line 105 where `headers` variable is initialized.
That way the argument passed to `openBitmapInputStream` would be pure Java/Kotlin class instead of ReactNative-specific class (`ReadableMap`)